ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/helpers/C_TypeHelper.java

```diff
@@ -36,11 +43,13 @@ private static C_Type createType(final String type, final boolean isArray, final
         }
 
         switch (type) {
+            case "java.lang.Boolean":
             case "boolean":
                 return new PrimitiveType("ds3_bool", isArray);
 
-            case "double":
+            case "java.lang.Double":
```

Comment:
Why is a Double treated at a uint64_t?  Shouldn't this be treated as a float instead?